### DISCUSSION TITLE
Tasmota, solarpanel and battery

### DISCUSSION THREAD

**Discussion by icho40** [arendst/Tasmota#21992] [Q&A] [answered]
Hello,

I hope I can get some help here.

My problem is a FireBeetle 2 ESP32-C6.
This is already designed for solar panel and battery operation.
Quote:
FireBeetle 2 ESP32-C6 integrates lithium battery charging management, allowing the lithium battery to be charged via Type-C, 5V DC, and solar panels
[Firebeetle 2](https://wiki.dfrobot.com/SKU_DFR1075_FireBeetle_2_Board_ESP32_C6)

I compiled a firmware with the TasmoCompiler from benzino99.
This was necessary to be able to connect a VL53L1X
[VL53L1x](https://tasmota.github.io/docs/VL53Lxx/)
A BME280 is also connected.

Everything works perfectly.

But when I connect a battery, only the red LED lights up. 
There are no values in the web interface or in the console.

How can I tell whether the battery is being charged and what the charge status is?
I would also like to see the values of the connected solar panel.
I would be grateful for any advice.

Regards
Willi

Translated with DeepL.com (free version)

**Comment by sfromis** [comment, marked as answer]
The page you posted a link to says (shown as connected to the + side of the battery connector):
> IO0: Battery voltage detection pin

If you config that for analog, Tasmota can get a number about the voltage.

And the "Charging indicator" LED next to USB on when charging, or off if not. It helps to actually read the board docs yourself :grinning:

**Reply by icho40** [reply]
I have read that, but I can't find an IO0 pin in the pin diagram.
also not in the web interface at
Settings -> Device settings

**Reply by sfromis** [reply]
The pin diagram clearly mentions how IO0 (gpio 0) is connected. Not broken out to the edge, as it is already taken up.
![image](https://github.com/user-attachments/assets/79a7eb8a-1253-4d42-ae06-f65b9a592f8c)

And when it comes to configuring, you clearly already have found out how to configure gpio pins, this is just one more gpio to configured.

**Reply by barbudor** [reply]
Where do you see such menu ? Tasmota doesn't have such menu

It's "Configuration" => "Configure Template"
There you should find (GP)IO0

**Reply by icho40** [reply]
@sfromis 
Sorry, I did not immediately realize that with “IO0”
meant the GPIO0.

@barbudor 
This is the Deepl.com translation of my german menu:
"Einstellungen->Geräteeinstellungen"

Thank you very much for your help.
Sometimes I just need a pat on the back of the head
